src/Config/Validator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Config;
+
+use Infection\Logger\ResultsLoggerTypes;
+use Symfony\Component\Filesystem\Exception\IOException;
+
+/**
+ * @internal
+ */
+final class Validator
```

Comment:
What about making this a `ConfigValidator`, which takes the infection config as its input, and does the validation for all other things as well?

Reply:
> which takes the infection config as its input, and does the validation for all other things as well?

it does exactly that. I'm not sure I got what you mean. It takes `InfectionConfig` instance and does validation (only `logs` key is validated at the moment)

If you are speaking about renaming `Validator` to `ConfigValidator` - I'm not sure, because it is already in `Config` namespace, so it would be `Infection\Config\ConfigValidator`.

Reply:
My bad, i didn't think about simply adding more statements to `validate` in the future, which works the same as having it in a constructor.